mainWidget/main_widget_FloorsAndRooms.yaml

```diff
@@ -1,139 +1,219 @@
+                                                        groupItemLabel: =loop.equipmentItem.label
+                                                        groupItemLocation: =loop.rooms.label
                             - component: oh-repeater
                               config:
-                                fetchMetadata: semantics,metadata,listWidget
-                                filter: '((loop.hvacModeItem.groupNames.indexOf(loop.hvacItem.name) >= 0) && (loop.hvacModeItem.type=="String")) ? true : false'
-                                for: hvacModeItem
-                                itemTags: Control,Temperature
+                                fetchMetadata: semantics,metadata,listWidget,uiSemantics
+                                filter: loop.radiatorItem.metadata.semantics.config.hasLocation == loop.rooms.name
+                                for: radiatorItem
+                                fragment: true
+                                itemTags: RadiatorControl
+                                sourceType: itemsWithTags
+                              slots:
+                                default:
+                                  - component: f7-col
+                                    config:
+                                      class:
+                                        - '=((vars.objVar.room == loop.radiatorItem.metadata.semantics.config.hasLocation || (vars.objVar.selectSection == "SECTION2" && !vars.objVar.floor && !vars.objVar.room) || (vars.objVar.selectSection == "SECTION2" && vars.objVar.floor == loop.rooms.metadata.semantics.config.isPartOf && !vars.objVar.room) || (vars.objVar.selectSection == "SECTION1")) && (vars.objVar.selectThing == "Climate" || !vars.objVar.selectThing )) ? "visible" : "invisible"'
+                                      large: 50
+                                      medium: 50
+                                      small: 90
+                                      style:
+                                        overflow: hidden
+                                      width: 100
+                                      xlarge: 33
+                                      xsmall: 100
+                                    slots:
+                                      default:
+                                        - component: widget:main_widget_RadiatorControl_Card_Dev
+                                          config:
+                                            Title: "=loop.radiatorItem.metadata.uiSemantics ? loop.radiatorItem.metadata.uiSemantics.config.equipment + loop.radiatorItem.metadata.uiSemantics.config.preposition + loop.radiatorItem.metadata.uiSemantics.config.location : loop.radiatorItem.label"
+                                            groupItem: =loop.radiatorItem.name
+                                            groupItemLabel: =loop.radiatorItem.label
+                                            groupItemLocation: =loop.rooms.label
```

Comment:
What is this used for ?

Reply:
Think I can answer myself from your description.

Reply:
Extensions for future widgets.
Location is available and label is present.
Maybe uiSemantics are not needed.

Reply:
I have to think about this, not really convinced to remove uiSemantics.

Reply:
Understood. Let me check this a bit more